src/Transport/Seam/MessagePump.cs

```diff
@@ -27,6 +28,11 @@ public Task Init(Func<PushContext, Task> pump, CriticalError criticalError, Push
             messagePump = pump;
             circuitBreaker = new RepeatedFailuresOverTimeCircuitBreaker("MessagePump", TimeSpan.FromSeconds(30), ex => criticalError.Raise("Failed to receive message from Azure Service Bus.", ex));
 
+            if (settings.PurgeOnStartup)
+            {
+                throw new ConfigurationErrorsException("Azure Service Bus transport doesn't support PurgeOnStartup behaviour");
```

Comment:
We shouldn't throw `ConfigurationErrorsException` for non config section related things. Maybe `InvalidOperationException`?